### PR TITLE
chore(j-s): Court Autofill

### PR DESCRIPTION
**Pull request by gudjong**
# Court Autofill

[Autofilla dómstól þegar farið er á málsmeðferðar skjá](https://app.asana.com/0/1199153462262248/1207539443130891/f)

## What

- Delays court autofill for indictment cases until the user enters the court selection screen. Previously, the court was autofilled during case creation.
- Court autofill for non indictment cases was not changed.

## Why

- The court should not be autofilled until the user is ready to select a court.

### Before court selection
image

### After court selection
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Improved handling of case state and court assignment based on case type.
  - Enhanced GraphQL query for user information with additional fields.

- **Bug Fixes**
  - Corrected conditions for indictment case handling and defender email updates.

- **Refactor**
  - Streamlined logic for setting case properties in multiple service files.

- **Tests**
  - Updated test cases to reflect new logic for court ID assignment.

### CODE REVIEW

**Review by unakb — APPROVED** [review]
Just the rabbit comments

File: apps/judicial-system/web/src/components/UserProvider/currentUser.graphql [hunk 1]
```diff
@@ -13,12 +13,10 @@ query CurrentUser {
     canConfirmIndictment
     institution {
       id
-      created
-      modified
-      name
       type
+      name
+      defaultCourtId
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/UserProvider/currentUser.graphql:18]: It is defined in the graphql schema.

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Processing/Processing.tsx [hunk 2]
```diff
@@ -35,14 +37,38 @@ import { strings } from './processing.strings'
...
+    }
+  }, [
+    refreshCase,
+    updateCase,
+    user?.institution?.defaultCourtId,
+    workingCase.court,
+    workingCase.id,
+  ])
+
+  useOnceOn(isCaseUpToDate, initialize)
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Processing/Processing.tsx:70]: No, thank you.

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Processing/Processing.tsx [hunk 3]
```diff
@@ -35,14 +37,38 @@ import { strings } from './processing.strings'
...
   const router = useRouter()
   const isTrafficViolationCaseCheck = isTrafficViolationCase(workingCase)
 
+  const initialize = useCallback(async () => {
+    if (!workingCase.court) {
+      await updateCase(workingCase.id, {
+        courtId: user?.institution?.defaultCourtId,
+      })
+      refreshCase()
+    }
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Processing/Processing.tsx:61]: The method `updateCase` has its own error handling so I prefer not adding error handling here.

**Pull request merged**